framework/src/transfers/MultiAppMeshFunctionTransfer.C

```diff
@@ -219,7 +219,8 @@ MultiAppMeshFunctionTransfer::execute()
   {
     if (i_proc == processor_id())
       continue;
-    _communicator.send(i_proc, outgoing_points[i_proc]);
+    std::vector<Parallel::Request> reqs(n_processors());
```

Comment:
In this scenario, especially since you are using blocking receives, it's probably not necessary to `wait()`.  But in cases where you might want to reuse the send buffer's data for some other purpose, it's undefined behavior to not `wait()` on the request before doing that.  Another scenario where you'd need to wait() would be when the send buffer could potentially go out of scope before the send completes.  Generally speaking, it's just good programming practice to wait on your requests, it squashes bugs before they ever have a chance to happen...